crates/circuit/src/dag_circuit.rs

```diff
@@ -0,0 +1,6474 @@
+        Ok(PyLegacyResources {
+            clbits: res.getattr("clbits")?.downcast_into_exact()?.unbind(),
+            cregs: res.getattr("cregs")?.downcast_into_exact()?.unbind(),
+        })
+    }
+
+    fn node_resources(&self, node: &Bound<PyAny>) -> PyResult<PyLegacyResources> {
+        let res = self.node_resources.bind(node.py()).call1((node,))?;
+        Ok(PyLegacyResources {
+            clbits: res.getattr("clbits")?.downcast_into_exact()?.unbind(),
+            cregs: res.getattr("cregs")?.downcast_into_exact()?.unbind(),
+        })
+    }
+}
+
+struct PyVariableMapper {
+    mapper: Py<PyAny>,
+}
+
+impl PyVariableMapper {
+    fn new(
+        py: Python,
+        target_cregs: Bound<PyAny>,
+        bit_map: Option<Bound<PyDict>>,
+        var_map: Option<Bound<PyDict>>,
+        add_register: Option<Py<PyAny>>,
+    ) -> PyResult<Self> {
+        let kwargs: HashMap<&str, Option<Py<PyAny>>> =
+            HashMap::from_iter([("add_register", add_register)]);
+        Ok(PyVariableMapper {
+            mapper: imports::VARIABLE_MAPPER
+                .get_bound(py)
+                .call(
+                    (target_cregs, bit_map, var_map),
+                    Some(&kwargs.into_py_dict_bound(py)),
+                )?
+                .unbind(),
+        })
+    }
+
+    fn map_condition<'py>(
+        &self,
+        condition: &Bound<'py, PyAny>,
+        allow_reorder: bool,
+    ) -> PyResult<Bound<'py, PyAny>> {
+        let py = condition.py();
+        let kwargs: HashMap<&str, Py<PyAny>> =
+            HashMap::from_iter([("allow_reorder", allow_reorder.into_py(py))]);
+        self.mapper.bind(py).call_method(
+            intern!(py, "map_condition"),
+            (condition,),
+            Some(&kwargs.into_py_dict_bound(py)),
+        )
+    }
+
+    fn map_target<'py>(&self, target: &Bound<'py, PyAny>) -> PyResult<Bound<'py, PyAny>> {
+        let py = target.py();
+        self.mapper
+            .bind(py)
+            .call_method1(intern!(py, "map_target"), (target,))
+    }
+
+    #[allow(dead_code)]
+    fn map_expr<'py>(&self, node: &Bound<'py, PyAny>) -> PyResult<Bound<'py, PyAny>> {
+        let py = node.py();
+        self.mapper
+            .bind(py)
+            .call_method1(intern!(py, "map_expr"), (node,))
+    }
+}
+
+#[pyfunction]
+fn reject_new_register(reg: &Bound<PyAny>) -> PyResult<()> {
+    Err(DAGCircuitError::new_err(format!(
+        "No register with '{:?}' to map this expression onto.",
+        reg.getattr("bits")?
+    )))
+}
+
+impl IntoPy<Py<PyAny>> for PyVariableMapper {
+    fn into_py(self, _py: Python<'_>) -> Py<PyAny> {
+        self.mapper
+    }
+}
+
+#[pyclass(module = "qiskit._accelerate.circuit")]
+#[derive(Clone, Debug)]
+struct BitLocations {
+    #[pyo3(get)]
+    pub index: usize,
+    #[pyo3(get)]
+    registers: Py<PyList>,
+}
+
+impl BitLocations {
+    fn set_index(&mut self, index: usize) {
+        self.index = index
+    }
+}
```

Comment:
This is a rust space setter method, it isn't exposing this to python. That being said `index` is pub so rust space code could directly set it. But I don't see the harm in having a method to do it like this. I'll add a docstring though to make it explictly a rust space setter.

Reply:
Actually the visibility here is really confusing to me, the field is public but the setter method isn't. I don't think we need any of it to be public because afaict this is only for python. I'll reorganize this a bit.

Reply:
I dropped the method completely and made `index` private since it's not used outside this module in: https://github.com/Qiskit/qiskit/pull/12550/commits/adf0c918c6594b6b743c164c01f5481823aa7e23